src/thing_doors.c

```diff
@@ -311,6 +311,29 @@ TbBool thing_is_deployed_door(const struct Thing *thing)
     return true;
 }
 
+TbBool thing_is_sellable_door(const struct Thing* thing)
+{
+    if (thing_is_invalid(thing))
+        return false;
+    if (thing->class_id != TCls_Door)
+        return false;
+    struct DoorConfigStats* doorst = &gameadd.trapdoor_conf.door_cfgstats[thing->model];
+    return (doorst->unsellable == 0);
+}
+
+struct Thing *get_door_for_slab_position(MapSlabCoord slb_x, MapSlabCoord slb_y)
+{
+    MapCoord pos_x = subtile_coord_center(slab_subtile_center(slb_x));
+    MapCoord pos_y = subtile_coord_center(slab_subtile_center(slb_y));
+    return get_door_around_of_model_and_owned_by(pos_x, pos_y, -1, -1);
+}
+
+TbBool slab_is_sellable_door(MapSlabCoord slb_x, MapSlabCoord slb_y)
+{
+    struct Thing* doortng = get_door_for_slab_position(slb_x, slb_y);
```

Comment:
It seems like you made this function, but we already have a `get_door_for_position`. You can get a stl from a slb with `slab_subtile_center`

Reply:
``struct Thing* doortng = get_door_for_position(slab_subtile_center(slb_x), slab_subtile_center(slb_y));``

like this?

Reply:
yes